backend/src/v5/processors/users.js

```diff
@@ -120,33 +117,24 @@ Users.resetPassword = async (user) => {
 };
 
 Users.getAvatar = async (username) => {
-	const avatarStream = await getAvatarStream(username);
-	const arrayBuffer = await avatarStream.arrayBuffer();
-	const fileExt = await fileExtensionFromBuffer(Buffer.from(arrayBuffer));
+	try {
+		const userId = await getUserId(username);
+		const avatarBuffer = await getUserAvatarBuffer(userId);
+		const fileExt = await fileExtensionFromBuffer(avatarBuffer);
 
-	return {
-		buffer: Buffer.from(arrayBuffer),
-		extension: fileExt || 'png',
-	};
+		return {
+			buffer: avatarBuffer,
+			extension: fileExt || 'png',
+		};
+	} catch (error) {
+		logger.logError(`Failed to fetch avatar from URL: ${error.message}`);
+
+		throw templates.unknown;
+	}
 };
 
 Users.uploadAvatar = async (username, avatarObject) => {
```

Comment:
Since you opted for the file path route, we need to make sure we delete the temorary file from the server after we're done (or if it errored)